apps/ledger-live-desktop/src/renderer/reducers/settings.ts

```diff
@@ -100,27 +107,35 @@ export type SettingsState = {
   vaultSigner: VaultSigner;
 };
 
-const DEFAULT_LANGUAGE_LOCALE = "en";
-export const getInitialLanguageLocale = (fallbackLocale: string = DEFAULT_LANGUAGE_LOCALE) => {
-  const detectedLanguage = getSystemLocale() || fallbackLocale;
-  return getLanguages().find(lang => detectedLanguage.startsWith(lang)) || fallbackLocale;
-};
-const DEFAULT_LOCALE = "en-US";
-export const getInitialLocale = () => {
-  const initialLanguageLocale = getInitialLanguageLocale();
-  return (
-    defaultLocaleForLanguage[initialLanguageLocale as keyof typeof defaultLocaleForLanguage] ||
-    DEFAULT_LOCALE
-  );
+export const getInitialLanguageAndLocale = (): { language: Language; locale: Locale } => {
+  const systemLocal = getSystemLocale();
+
+  // Find language from system locale (i.e., en, fr, es ...)
+  const languageId = LanguageIds.find(lang => systemLocal.startsWith(lang));
+
+  // If language found, try to find corresponding locale
+  if (languageId) {
+    // const localeId = Languages[languageId].locales.find(lang => systemLocal.startsWith(lang));
+    // TODO Hack because the typing on the commented line above doesn't work
+    const languageLocales = Languages[languageId].locales as Locales;
```

Comment:
If I define 
`export const Languages: LanguageMap<LanguageDefinition> = {} as const` I can avoid this hack
Could you check if it satisfies what you expect, please?

When I do the change I have relevant errors as follows:
```

 Running typescript type checker on 1657 files…
apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/steps/restore/Language.tsx:29:5 - error TS18048: 'DEFAULT_LANGUAGE.deviceSupport' is possibly 'undefined'.

29     DEFAULT_LANGUAGE.deviceSupport.label,
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
apps/ledger-live-desktop/src/renderer/screens/settings/sections/General/ChangeDeviceLanguagePromptDrawer.tsx:80:15 - error TS18048: 'DEFAULT_LANGUAGE.deviceSupport' is possibly 'undefined'.

80               DEFAULT_LANGUAGE.deviceSupport.label
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

That makes sense as we define 
` deviceSupport: { label: DeviceLanguages; id: number } | undefined;`